src/iota/essence.c

```diff
@@ -483,27 +467,14 @@ static uint16_t essence_sign_signature_unlock_block(
     uint32_t signature_length = 0;
 
     uint8_t ret = 0;
-    BEGIN_TRY
-    {
-        TRY
-        {
-            // create key pair and convert pub key to bytes
-            ret = ed25519_get_key_pair(bip32_path, BIP32_PATH_LEN, &pk, &pub);
-
-            ret = ret && ed25519_sign(&pk, essence_hash, BLAKE2B_SIZE_BYTES,
-                                      pBlock->signature, &signature_length);
-        }
-        CATCH_OTHER(e)
-        {
-            THROW(e);
-        }
-        FINALLY
-        {
-            // always delete from stack
-            explicit_bzero(&pk, sizeof(pk));
-        }
-    }
-    END_TRY;
```

Comment:
if an exception is throw `pk` would not be wiped from memory. Same explanation as above